gutenberg-prototype.php

```diff
@@ -549,32 +542,51 @@ public function get_plugin_info( $response, $action, $args ) {
 		 *
 		 * @access public
 		 * @global $wp_filesystem
-		 * @param  string           $source        File source location.
-		 * @param  string           $remote_source Remote file source location.
-		 * @param  WP_Upgrader      $upgrader      WP_Upgrader Upgrader instance.
+		 * @param  string           $source        File source location
+		 * @param  string           $remote_source Remote file source location
+		 * @param  WP_Upgrader      $upgrader      WP_Upgrader instance
+		 * @param  array            $hook_extra    Data regarding plugin being updated.
 		 * @return string|WP_Error
 		 */
-		public function upgrader_source_selection( $source, $remote_source, $upgrader ) {
+		public function upgrader_source_selection( $source, $remote_source, $upgrader, $hook_extra ) {
 			global $wp_filesystem;
 
-			if ( strstr( $source, '/gutenberg-' ) ) {
-				$corrected_source = trailingslashit( $remote_source ) . $this->config['proper_folder_name'] . '/';
+			if ( ! isset( $hook_extra['plugin'] ) || $this->config['plugin_file'] !== $hook_extra['plugin'] ) {
+				return $source;
+			}
 
-				$upgrader->skin->feedback( sprintf(
-					__( 'Renaming %s to %s&#8230;', 'gutenberg-prototype' ), '<span class="code">' . basename($source) . '</span>', '<span class="code">' . $this->config['proper_folder_name'] . '</span>'
-				) );
+			if ( $this->config['release_asset'] ) {
```

Comment:
Can we detect a release asset more efficiently? Setting a variable to true will always run this condition and ignore the other.

Reply:
Well, we are not focusing on supporting other updates. Would like to keep the code clean and focused on what I am trying to achieve for this project only. It's like saying you have two good kidneys but here is a third just in case they both fail.

Reply:
Well, we are not focusing on supporting other updates. Would like to keep the code clean and focused on what I am trying to achieve for this project only. It's like saying you have two good kidneys but here is a third just in case they both fail.

Reply:
Sorry, GitHub.com was having issues.
I’ll fix the PR accordingly.